models/yang/sonic/sonic-bgp-common.yang

```diff
@@ -0,0 +1,362 @@
+      }
+      enum none {
+        description "No community attribute";
+      }
+    }
+    description
+      "type describing variations of community attributes";
+  }
+
+  typedef sonic_bgp_direction {
+    type enumeration {
+      enum inbound {
+        description
+          "Refers to information received from the BGP peer";
+      }
+      enum outbound {
+        description
+          "Refers to information advertised to the BGP peer";
+      }
+    }
+    description
+        "Type to describe the direction";
+  }
+
+
+  grouping sonic-bgp-cmn {
+
+    leaf vrf_name {
+      type string;
+        description "Network-instance/VRF name";
+    }
+
+    leaf peer_group_name {
+      type string;
+        description "Peer group name";
+    }
+
+    leaf local_asn {
+      type uint32;
+        description "Local AS number";
+    }
+
+    leaf description {
+      type string;
+    }
+
+    leaf peer_asn {
+      type uint32;
+        description "Peer AS number";
+    }
+
+    leaf peer_type {
+      type bgp_peer_type;
+    }
+
+    leaf ebgp_multihop {
+      type boolean;
+        description "EBGP Multihop enabled";
+    }
+
+    leaf ebgp_multihop_ttl {
+      type uint8 {
+        range "1..255";
+      }
+        description "EBGP Multihop TTL";
+    }
+
+    leaf auth_password {
+      type string;
+        description "Authuntiation password";
+    }
+
+    leaf keepalive_interval {
+      type uint16 {
+        range "1..3600";
+      }
+        description "Keepalive interval";
+    }
+
+    leaf hold_time {
+      type uint16 {
+        range "1..3600";
+      }
+        description "Hold time";
+    }
+
+    leaf conn_retry {
+      type uint16 {
+        range "1..3600";
+      }
+        description "Connection retry time";
+    }
+
+    leaf min_adv_interval {
+      type uint16 {
+        range "1..3600";
+      }
+        description "minium advertisement interval";
+    }
+
+    leaf local_address {
+      type inet:ip-address;
+        description "Local address to use for connection";
+    }
+
+    leaf passive_mode {
+      type boolean;
+        description "Passive mode";
+    }
+
+    leaf capability_ext_nexthop {
+      type boolean;
+        description "Advertise extended nexthop capability";
+    }
+
+    leaf disable_ebgp_connected_route_check {
+      type boolean;
+        description "Connected checked disabled";
+    }
+
+    leaf enforce_first_as {
+      type boolean;
+        description "Enforce first AS for EBGP";
+    }
+
+    leaf solo_peer {
+      type boolean;
+        description "Part of own update group";
+    }
+
+    leaf ttl_security_hops {
+      type uint8;
+        description "TTL";
+    }
+
+    leaf bfd {
+      type boolean;
+        description "BFD status";
+    }
+  }
+
+
+  grouping sonic-bgp-cmn-af {
+
+    leaf vrf_name {
+        type string;
+        description "Network-instance/VRF name";
+    }
+
+    leaf afi_safi {
+        type string;
```

Comment:
Should this be enum ?

Reply:
yes, will add the various address family names in enum.